ComPact/Verifier/PactVerifier.cs

```diff
@@ -70,6 +72,15 @@ public async Task VerifyPactAsync(string path)
             if (_config.PublishVerificationResults)
             {
                 await PublishVerificationResultsAsync(pact, tests, _config.ProviderVersion, _config.PactBrokerClient, publishVerificationResultsUrl);
+
+                if (_config.ProviderTags != null)
+                {
+                    var tags = _config.ProviderTags.ToList();
+                    if (tags.Any())
```

Comment:
You could merge these two ifs with `_config.ProviderTags?.FirstOrDefault != null`

Reply:
Because the type is IEnumerable, you should only do something which causes enumeration once since some implementation of the IEnumerable interface will fail if you try multiple enumerations... though that said, I could do:
```
var tags = _config.ProviderTags?.ToList();
if (tags?.FirstOrDefault() != null)
{
...
```

which reduces it to a single if whilst still allowing the type to be IEnumerable...

Update coming shortly :)